### PR TITLE
load malaria data

### PR DESCRIPTION
**Pull request by jhung0**
load training, validation, test data as a list of dictionaries

### CODE REVIEW

**Comment by jhung0** [comment]
#14

**Comment by JihongJu** [comment]
@jhung0 What do you think to organize the datasets as

```Python
from keras_rcnn.datasets import malaria, pascal, coco
train_data, test_data, validate_data = malaria.load_data()
train_data, test_data = pascal.load_data()
# and coco is alike.
```

**Comment by 0x00b1** [comment]
@JihongJu I agree! 👍

File: tests/datasets/test_malaria.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from __future__ import print_function
+import pytest
+from keras.datasets import malaria
+
+training, test = malaria.load_data()
```
Comment (0x00b1) [inline tests/datasets/test_malaria.py]: Wrap this inside a function, e.g.

```Python
def test_load_data():
    training, test = malaria.load_data()

    assert len(training) == 1208
    assert len(test) == 120 
```

File: tests/datasets/test_malaria.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from __future__ import print_function
+import pytest
+from keras.datasets import malaria
```
Comment (0x00b1) [inline tests/datasets/test_malaria.py]: `import keras_rcnn.datasets.malaria`

File: tests/datasets/test_malaria.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from __future__ import print_function
+import pytest
```
Comment (0x00b1) [inline tests/datasets/test_malaria.py]: Unneeded

File: tests/datasets/test_malaria.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from __future__ import print_function
```
Comment (0x00b1) [inline tests/datasets/test_malaria.py]: Unneeded

File: keras_rcnn/datasets/malaria.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import json
+import os.path
+
+import keras.utils.data_utils
+
+
+def load_data():
+    """Loads the malaria dataset.
```
Comment (0x00b1) [inline keras_rcnn/datasets/malaria.py]: You could probably provide a more descriptive summary of the data. 😆

File: keras_rcnn/datasets/malaria.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    # Returns
```
Comment (0x00b1) [inline keras_rcnn/datasets/malaria.py]: Yes the following docstring format:

```Python
    """

    :param bar: 
    
    :return: 
    """
```

File: keras_rcnn/datasets/malaria.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    filename = os.path.join(pathname, "training.json")
+
+    with open(filename) as data:
+        training = json.load(data)
+
+    filename = os.path.join(pathname, "test.json")
+
+    with open(filename) as data:
+        test = json.load(data)
+
```
Comment (0x00b1) [inline keras_rcnn/datasets/malaria.py]: You’ll need to iterate across both the training and test dictionaries and modify the “filename” values:

```Python
for dictionary in training:
    dictionary["filename"] = os.path.join(pathname, dictionary["filename"])

for dictionary in test:
    dictionary["filename"] = os.path.join(pathname, dictionary["filename"])
```

File: tests/datasets/test_malaria.py [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import keras_rcnn.datasets.malaria
+
+def test_load_data():
+    training, test = keras_rcnn.datasets.malaria.load_data()
+
+    assert len(training) == 1208
+    assert len(test) == 120
```
Comment (0x00b1) [inline tests/datasets/test_malaria.py]: 👍

Comment (0x00b1) [inline tests/datasets/test_malaria.py]: I’d also validate whether the value of a filename (e.g. from the first dictionary of training and test) exists.

File: tests/datasets/test_malaria.py [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import keras_rcnn.datasets.malaria
+
+def test_load_data():
+    training, test = keras_rcnn.datasets.malaria.load_data()
```
Comment (0x00b1) [inline tests/datasets/test_malaria.py]: We should cache the `~/.keras` directory on Travis to minimize re-downloading.

https://docs.travis-ci.com/user/caching/#Arbitrary-directories

**Comment by 0x00b1** [comment]
@jhung0 Add yourself to AUTHORS.rst

File: keras_rcnn/datasets/malaria.py (hunk 7 above)
Comment (jhung0) [inline keras_rcnn/datasets/malaria.py]: Even with the filenames I have in the malaria-data.tar.gz file now? I changed it to the absolute path.

File: tests/datasets/test_malaria.py (hunk 9 above)
Comment (jhung0) [inline tests/datasets/test_malaria.py]: like 
```
cache: pip
  directories:
    - ~/.keras
```

Comment (0x00b1) [inline tests/datasets/test_malaria.py]: Yep

**Comment by jhung0** [comment]
Check again?

**Comment by jhung0** [comment]
#14

File: keras_rcnn/datasets/malaria.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    Loads the malaria dataset.
+    It is made up of images of blood smears from different patients taken with different microscope cameras.
+    Each image has a collection of bounding box coordinates and corresponding class labels for each cell present.
+    :return:
+        Tuple of list of dictionaries where each dictionary corresponds to an image: `training, test`.
+    """
+    origin = "http://keras-rcnn.storage.googleapis.com/malaria.tar.gz"
+
+    pathname = keras.utils.data_utils.get_file("malaria", origin=origin, untar=True)
```
Comment (JihongJu) [inline keras_rcnn/datasets/malaria.py]: Wow, this is cool. 👍

File: keras_rcnn/datasets/create_malaria.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+            x['bboxes'] = get_data(label_file)
+
+            dictionary_t.append(x)
+            
+        dictionary.append(dictionary_t)
+
+    return dictionary
+
+data = load_data()
+with open('/home/jhung0/malaria-data.pickle', 'wb') as f:
```
Comment (JihongJu) [inline keras_rcnn/datasets/create_malaria.py]: Just to recommend, maybe you want to hide your home directory using `os.path.expanduser("~")`?

**Pull request merged**